### PR TITLE
Add UI for updating submission metadata.

### LINKED ISSUE #1176: Add feature to update submission metadata.
**Current Behaviour:** There is no feature to update the submission meta once a submission is done by the participant.
**Expected Behaviour:**
The Participant should be able to update:

- [x] Method Name

- [x] Method Description

- [x] Project Url

- [x] Publication Url

### PR DESCRIPTION
**Pull request by RishabhJain2018**
fixes #1176. @deshraj @aka-jain Please review the PR.
1.
screen1
2. 
screen2
3. 
screen3
4.
screen4

### CODE REVIEW

File: frontend/src/css/modules/submission.scss [hunk 1]
```diff
@@ -7,3 +7,11 @@
 .time {
 	position: static;
 }
+
+.pointer {
```
Comment (aka-jain) [inline frontend/src/css/modules/submission.scss]: I think there is already a class present named pointer. Please recheck again in base.scss.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -984,6 +984,101 @@
...
+                    +                                   'type="submit" value="Submit">Submit</button>'
+                    +                           '</div>'
+                    +                       '</form>'
+                    +                   '</div>'
+                    +               '</div>'
+                    +           '</div>'
+                    +   '</section>';
+
+        vm.showMdDialog = function(ev, submissionId) {
+            console.log(submissionId);
```
Comment (aka-jain) [inline frontend/src/js/controllers/challengeCtrl.js]: Remove console please.

File: frontend/src/views/web/challenge/my-submission.html [hunk 3]
```diff
@@ -42,27 +42,34 @@
                 </div>
                 <div class=" ev-card-body exist-team-card">
                     <div class="row">
-                        <div class="col s12">
+                        <div class="col s12 all-submission-table-scroll">
```
Comment (aka-jain) [inline frontend/src/views/web/challenge/my-submission.html]: Make a separate div for this class inside col div. Don't use own classes with grid related classes like col, m2 etc.

**Comment by aka-jain** [comment]
@RishabhJain2018 please check comments, fix them and rest looks good to me.

**Comment by RishabhJain2018** [comment]
@aka-jain Please review.

**Comment by deshraj** [comment]
@RishabhJain2018 can you please add tests for the updated endpoint? The code looks good to me though.

File: apps/jobs/urls.py [hunk 4]
```diff
@@ -3,9 +3,9 @@
 from . import views
 
 urlpatterns = [
-    url(r'challenge/(?P<challenge_id>[0-9]+)/'
```
Comment (taranjeet) [inline apps/jobs/urls.py]: I am really not able to infer any change in `jobs` app. The only valid thing that I could infer was adding support for `PUT` method, but there is no action for the same method. 

If that is the case(PUT method), please only add that change in this PR and the cleaning and updating part can go in other PR?


Also are we not following of creating different PR for frontend and backend changes?

File: frontend/src/js/controllers/challengeCtrl.js [hunk 5]
```diff
@@ -984,6 +984,100 @@
...
+                    +                   '</div>'
+                    +               '</div>'
+                    +           '</div>'
+                    +   '</section>';
+
+        vm.showMdDialog = function(ev, submissionId) {
+            for (var i=0;i<vm.submissionResult.count;i++){
+                if (vm.submissionResult.results[i].id === submissionId) {
+                    vm.submissionMetaData = vm.submissionResult.results[i];
+                    console.log(vm.submissionMetaData);
```
Comment (taranjeet) [inline frontend/src/js/controllers/challengeCtrl.js]: Is this console statement necessary here ?

Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: No, I will remove it.

File: apps/jobs/urls.py (hunk 4 above)
Comment (RishabhJain2018) [inline apps/jobs/urls.py]: Yes, I just added it for future use. Morover, I will maintain the different PR's for frontend and backend :+1: .
I also added the test cases regarding the view function `change_submission_data_and_visibility`.
I will also add it in different PR.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 6]
```diff
@@ -984,6 +984,99 @@
         }
         };
 
+        var template = '<section class="ev-md-container text-center rm-overflow-y">'
```
Comment (deshraj) [inline frontend/src/js/controllers/challengeCtrl.js]: Do you really want to do it in this ugly manner? Cannot we have it in a view please?

Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeCtrl.js]: @deshraj Yes sure. I will create a separate file for it.

**Comment by deshraj** [comment]
@RishabhJain2018 please resolve the conflicts.

**Comment by RishabhJain2018** [comment]
@deshraj Please review.

**Pull request merged**